### PR TITLE
Win32 findfile fixes: Add support for Git for Windows 2 and other git.exe installations

### PR DESCRIPTION
**Pull request by csware**
Resurrecting #5150
Updated #5773

In-between the initial PR and now there have been several changes to Git for Windows >= 2:
* up to 2.24 an architecture specific subfolder was necessary for the etc and usr folders
* starting from 2.24 the etc folder is located directly in the installation folder but the usr folder is still in an architecture specific subfolder

### CODE REVIEW

File: src/win32/findfile.c [hunk 1]
```diff
@@ -183,6 +232,12 @@ int git_win32__find_system_dirs(git_str *out, const wchar_t *subdir)
 			&buf, HKEY_CURRENT_USER, REG_MSYSGIT_INSTALL_LOCAL, subdir) && buf.size)
 		git_str_join(out, GIT_PATH_LIST_SEPARATOR, out->ptr, buf.ptr);
 
+#ifdef _WIN64
```
Comment (ethomson) [inline src/win32/findfile.c:235]: Is this defined by mingw also, or just MSVC?

Comment (csware) [reply src/win32/findfile.c:235]: What would be the alternative for mingw?

Comment (csware) [reply src/win32/findfile.c:235]: Could we write `#if defined(_WIM64) || defined(__MINGW64__)`?

Comment (ethomson) [reply src/win32/findfile.c:235]: I think that we can just do `GIT_ARCH_64` here - this file will only ever be included for Windows.

Comment (csware) [reply src/win32/findfile.c:235]: Fixed.

**Comment by ethomson** [comment]
Thanks!  I merged this manually.

**Pull request closed without merging**